.github/workflows/build-types.yml

```diff
@@ -0,0 +1,84 @@
+name: Build types
+
+on:
+  push:
+    branches:
+      - dev
+    paths:
+      - '.github/workflows/build-types.yml'
+
+      # We don't need to hunt for every possible case, these should cover almost all cases
+      - 'ui/package.json'
+      - 'ui/build/build.types.js'
+      - 'ui/build/build.javascript.js'
+
+      # Manually authored .d.ts files
+      - 'ui/types/**'
+      # JSON API files
+      - 'ui/src/**/*.json'
+      - '!ui/src/api-file-example.json'
+  pull_request:
+    types:
+      - opened
+      - synchronize
+    paths:
+      - '.github/workflows/build-types.yml'
+
+      # We don't need to hunt for every possible case, these should cover almost all cases
+      - 'ui/package.json'
+      - 'ui/build/build.types.js'
+      - 'ui/build/build.javascript.js'
+
+      # Manually authored .d.ts files
+      - 'ui/types/**'
+      # JSON API files
+      - 'ui/src/**/*.json'
+      - '!ui/src/api-file-example.json'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
Yes, for the same workflow on the same ref. Ref is the PR's branch(_e.g. `someuser/quasar/patch-1`_) for `pull_request` events and `dev` for `push` events. So, it also applies if we quickly merge multiple PRs one after another. If we didn't specify `cancel-in-progress`, then it would just wait for the active ones to complete before starting.